scanpy/get.py

```diff
@@ -253,7 +253,7 @@ def var_df(
     # add obs values
     if len(obs_names) > 0:
         X = _get_obs_rep(adata, layer=layer)
-        matrix = X[adata.obs_names.get_indexer(obs_names), :]
+        matrix = X[adata.obs.index.isin(obs_names), :]
```

Comment:
Will these be in the right order?

Won't the value from `X` be in whatever order the appear, while the values in `obs_names` are in the order they were passed?

Reply:
This has made me realize some stuff needs to get fixed in `anndata`, mostly around `raw`. In future, I think this should look like `adata[obs_names].to_df(layer=layer)`, but we're not quite there yet.

Here's what you can do to make backed mode work for now:

```python
idxs = adata.obs_names.get_indexer(obs_names)
idxs_order = np.argsort(idxs_order)
matrix = X[idxs[idxs_order], :][np.argsort(idxs_order)]
```

In the next major release of anndata I'm thinking we should export some of the utilities that are used for giving all these indexing operations a consistent interface.

Reply:
Yes. That is why this should be safe for backed mode. Before the result is returned the columns are reorder to match the given order.